runtime/common/src/main/kotlin/kotlinx/serialization/modules/SerialModuleExtensions.kt

```diff
@@ -0,0 +1,83 @@
+@file:Suppress("RedundantVisibilityModifier")
+/*
+ * Copyright 2018 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kotlinx.serialization.modules
+
+import kotlinx.serialization.*
+import kotlin.reflect.KClass
+
+/**
+ * Returns a dependent serializer associated with a given reified type.
+ *
+ * This method is used in context-sensitive operations
+ * on a property marked with [ContextualSerialization], by a [ContextSerializer]
+ */
+public inline fun <reified T: Any> SerialModule.get(): KSerializer<T>? = get(T::class)
+
+/**
+ * Returns a serializer associated with KClass which given [value] has.
+ *
+ * This method is used in context-sensitive operations
+ * on a property marked with [ContextualSerialization], by a [ContextSerializer]
+ */
+@Suppress("UNCHECKED_CAST")
+public fun <T: Any> SerialModule.getByValue(value: T): KSerializer<T>? {
+    val klass = value::class
+    return get(klass) as? KSerializer<T>
+}
+
+@ImplicitReflectionSerializer
+public fun <T: Any> SerialModule?.getOrDefault(klass: KClass<T>) = this?.let { get(klass) } ?: klass.serializer()
+
+@Suppress("UNCHECKED_CAST")
+@ImplicitReflectionSerializer
+public fun <T: Any> SerialModule?.getByValueOrDefault(value: T): KSerializer<T> = this?.let { getByValue(value) } ?: value::class.serializer() as KSerializer<T>
+
+
+/**
+ * Composes [this] module with [other].
+ *
+ * If some serializer present in both modules, an [SerializerAlreadyRegisteredException] is thrown.
+ * To overwrite serializers, use [SerialModule.overwriteWith] function.
+ */
+public operator fun SerialModule.plus(other: SerialModule): SerialModule = SerializersModule {
+    +this@plus
```

Comment:
[opinated] this body is a good indicator that `+` is not very readable.

But 
```
this@plus.dumpTo(this)
other.dumpTo(this)
```
is not as well.

What about (maybe internal) extension `accept`? 
```
accept(this@plus)
accept(other)
```

Reply:
I suggest `include` – it is obvious in user code, too

Reply:
`include` is even better, agree